lib/spack/docs/packaging_guide.rst

```diff
@@ -2071,19 +2150,15 @@ The parameters are as follows:
     targets into.  It acts like a string, but it's actually its own
     special type, :py:class:`Prefix <spack.util.prefix.Prefix>`.
 
-``spec`` and ``prefix`` are passed to ``install`` for convenience.
-``spec`` is also available as an attribute on the package
-(``self.spec``), and ``prefix`` is actually an attribute of ``spec``
-(``spec.prefix``).
+The arguments ``spec`` and ``prefix`` are passed only for convenience, as they always
+correspond to ``self.spec`` and ``self.spec.prefix`` respectively.
```

Comment:
Personally, I think all methods should provide `self`, `spec`, and `prefix`. They aren't always needed, but for the case of `configure_args`, they usually are. The first line of pretty much every `configure_args` in Spack right now looks like `spec = self.spec`.

I am also fine with @tgamblin's proposal which would allow all of the following to work:
```python
def install(self):
def install(self, spec):
def install(self, spec, prefix):
```